modules/dnn/perf/perf_einsum.cpp

```diff
@@ -0,0 +1,121 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "perf_precomp.hpp"
+
+namespace opencv_test {
+
+struct EinsumParams {
+    int inputSize;
+    int outputSize;
+    std::string equation;
+    std::vector<MatShape> einsumInpShapes;
+    EinsumParams(std::string equation_, int inputSize_, int outputSize_,  std::vector<MatShape> einsumInpShapes_ = std::vector<MatShape>())
+    {
+        inputSize = inputSize_;
+        outputSize = outputSize_;
+        equation = equation_;
+        einsumInpShapes = einsumInpShapes_;
+    }
+};
+
+static inline void PrintTo(const EinsumParams& params, ::std::ostream* os) {
+     (*os) << "\nEqiation: \t" << params.equation << "\n"
+        << "InputSize: \t" << params.inputSize << "\n"
+        << "OutputSize: \t" << params.outputSize << "\n";
+
+        for(int i = 0; i < params.einsumInpShapes.size(); i++)
+        {
+            (*os) << "InputShape " << i << ": \t";
+            for(int j = 0; j < params.einsumInpShapes[i].size(); j++)
+            {
+                (*os) << params.einsumInpShapes[i][j] << " ";
+            }
+            (*os) << std::endl;
+        }
+}
+
+// test cases
+static const EinsumParams testEinsumConfigs[] = {
+    // TODO: Add tests with one input after ellips merge
+    {"ij, jk -> ik", 2, 1,  {{2, 3}, {3, 2}}},
+    {"ij, jk -> ik", 2, 1,  {{20, 30}, {30, 20}}},
+    {"ij, jk -> ik", 2, 1,  {{200, 300}, {300, 200}}},
+
+    {"imkj, injs -> imnks", 2, 1,  {{1, 4, 7, 9}, {1, 5, 9, 8}}},
+    {"imkj, injs -> imnks", 2, 1,  {{1, 4, 70, 90}, {1, 5, 90, 80}}},
+    {"imkj, injs -> imnks", 2, 1,  {{1, 4, 700, 900}, {1, 5, 900, 800}}},
+
+    {"ij -> i", 1, 1, {{30, 40}}},
+    {"ij -> i", 1, 1, {{300, 400}}},
+
+    {"...ij -> ...i", 1, 1, {{30, 40}}},
+    {"...ij -> ...i", 1, 1, {{300, 400}}},
+
+    {"...ij, ...jk -> ...ik", 2, 1, {{40, 50}, {50, 80}}},
+    {"...ij, ...jk -> ...ik", 2, 1, {{400, 500}, {500, 800}}}
```

Comment:
It makes sense to add couple of odd sizes like 127 or 113. It's usually hardest case for optimization and it shows the worst scenario.